app/lib/frontend/handlers.dart

```diff
@@ -154,7 +154,9 @@ Future<shelf.Response> _indexHandler(
     return redirectResponse(
         request.requestedUri.replace(path: newPath).toString());
   }
-  String pageContent = await backend.uiPackageCache?.getUIIndexPage(platform);
+  final isProd = _isProd(request);
```

Comment:
Idea: pondering if some sort of `withCache(..., Future<String> action(), {bool skipCache})`  helper would be useful here to keep us honest about get/set alignment, etc.